apps/emqx/src/emqx_session.erl

```diff
@@ -818,10 +834,13 @@ replay(ClientInfo, Session = #session{inflight = Inflight}) ->
         {ok, More, NSession} -> {ok, lists:append(Pubs, More), NSession}
     end.
 
--spec terminate(emqx_types:clientinfo(), Reason :: term(), session()) -> ok.
-terminate(ClientInfo, Reason, Session) ->
+-spec terminate(emqx_types:clientinfo(), emqx_types:conninfo(), Reason :: term(), session()) -> ok.
+terminate(ClientInfo, ConnInfo, Reason, Session) ->
     run_terminate_hooks(ClientInfo, Reason, Session),
     maybe_redispatch_shared_messages(Reason, Session),
+    % FIXME: unclear about the purpose of this
```

Comment:
Since this came from [`emqx_channel`](https://github.com/emqx/emqx/pull/11215/files#diff-27c0844d0e2116cd96569c34369a4b485d68529f0d21cb075c19f8bc5a59f477L1455), _it seems like_ the purpose was the "update" (`persist_if_session` > `emqx_persistent_session:persist`) to actually delete the stored session, since the `expired` comes from the channel timer and, by the time it arrives here, `persistent_session_status` should hopefully return `expired` and [discard](https://github.com/emqx/emqx/pull/11215/files#diff-1e3ee68c0cbaa4e519c54dd63eee5946ed745bd02c2fc03fffb9bf89d889f127L193) the data.  :thinking: